Add product vision (#119)
flowforge/handbook: product/index.md

This covers how we run our business and deliver service to our customers.

## Product Vision

> Our target customers are users familiar with Node-RED looking to run multiple instances or share access across multiple users.

Reviewer: ```suggestion
FlowForge's target customers are familiar with Node-RED looking to run Node-RED in production with confidence, from single users with a single instance to multiple instances across multiple users.
```

Same vibe, but a consultant starting out alone I think should be included in the target?

> We make it easy to create instances of Node-RED and to share configuration and work between them. We provide collaboration functionality allowing multiple people in a team to build and maintain a flow. 

Reviewer: Can we replace "We" with FlowForge in these changes?

> We give user a management platform that delivers the features around running node-red such that it requires minimal effort on the part of the customer to configure, update, and maintain their environments.

Reviewer: What FlowForge already provides and what we are building towards isn't differentiate enough I think for people unfamiliar with our product. Maybe it should describe what we offer now, and where we're building towards?

Also: s/node-red/Node-RED/

Author: As a vision though it should be forward looking rather than a product spec of what is currently offered, details of the current product are better suited to a product page on the website

> We enable customers to run the flows wherever they require based on what systems and devices they are connecting to.

Reviewer: More generally; where ever Node-RED runs, FlowForge will support you? Going back to @knolleary's remark yesterday that FlowForge is the answer to running Node-RED in production.

Author: This point was a specific reference to our capability for both server and device side instances, I like the point that its about what you are connecting to that determines where you need to run Node-RED

FlowForge is best placed deliver this because our team has extensive experience with the core of Node-RED.

> We differentiate from the competition by offering a range of deployment options for both public shared SaaS and dedicated instances either on-prem or in the customers cloud. Keeping our Node-RED experience consistent with the core Node-RED experience, where there are improvements we feed those back into the project rather than creating custom forks.

Reviewer: ```suggestion
We differentiate from the competition by offering a range of deployment options for both public shared SaaS and dedicated instances either on-prem or in the customers cloud. Keeping our Node-RED experience consistent with the core Node-RED experience. Improvements are fed back into the Node-RED project, rather than creating a fork of the Node-RED project.
```
